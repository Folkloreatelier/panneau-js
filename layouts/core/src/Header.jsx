import React, { Component } from 'react';
import PropTypes from 'prop-types';
import isObject from 'lodash/isObject';
import { PropTypes as PanneauPropTypes, withUrlGenerator } from '@panneau/core';

import Navbar from './Navbar';

import styles from './styles/header.scss';

const propTypes = {
    urlGenerator: PanneauPropTypes.urlGenerator,
    title: PropTypes.string,
    titleLink: PropTypes.string,
    navbar: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.object,
    ]),
    gotoHome: PropTypes.func.isRequired,
    gotoLink: PropTypes.func.isRequired,
    gotoRoute: PropTypes.func.isRequired,
    onNavbarClickTitle: PropTypes.func,
    onNavbarClickItem: PropTypes.func,
};

const defaultProps = {
    urlGenerator: null,
    title: 'Panneau',
    titleLink: '/',
    navbar: true,
    onNavbarClickTitle: null,
    onNavbarClickItem: null,
};

class Header extends Component {
    constructor(props) {
        super(props);

        this.onNavbarClickTitle = this.onNavbarClickTitle.bind(this);
        this.onNavbarClickItem = this.onNavbarClickItem.bind(this);
    }

    onNavbarClickTitle(...args) {
        const { onNavbarClickTitle } = this.props;
        if (onNavbarClickTitle !== null) {
            onNavbarClickTitle(...args);
        }
    }

    onNavbarClickItem(...args) {
        const { onNavbarClickItem } = this.props;
        if (onNavbarClickItem !== null) {
            onNavbarClickItem(...args);
        }
    }

    render() {
        const {
            urlGenerator,
            titleLink,
            navbar,
            onNavbarClickTitle,
            onNavbarClickItem,
            gotoHome,
            gotoLink,
            gotoRoute,
            ...props
        } = this.props;

        return (
            <div className={styles.container}>
                { navbar !== false ? (
                    <div className={styles.navbar}>
                        <Navbar
                            titleLink={urlGenerator !== null ? urlGenerator.route('home') : titleLink}
                            {...props}
                            {...(isObject(navbar) ? navbar : null)}
                            gotoHome={gotoHome}
                            gotoLink={gotoLink}
                            gotoRoute={gotoRoute}
                            onClickTitle={this.onNavbarClickTitle}
                            onClickItem={this.onNavbarClickItem}
                        />
                    </div>
                ) : null }
            </div>
        );
    }
}

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default withUrlGenerator()(Header);
