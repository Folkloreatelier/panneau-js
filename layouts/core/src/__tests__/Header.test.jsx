import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import { MemoryRouter } from 'react-router';
import Header from '../Header';

test('match snapshot', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Header gotoHome={() => {}} gotoLink={() => {}} gotoRoute={() => {}} />
        </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
