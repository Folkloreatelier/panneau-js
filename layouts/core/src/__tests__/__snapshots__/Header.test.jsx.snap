// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="#"
        onClick={[Function]}
      >
        Panneau
      </a>
      <button
        aria-expanded="false"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Menu
        </span>
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
      >
        <ul
          className="navbar-nav mr-auto"
        />
        <ul
          className="navbar-nav mr-auto navbar-right"
        />
      </div>
    </nav>
  </div>
</div>
`;
